webapp/packages/plugin-sql-editor/src/SqlResultTabs/SqlResultTabs.tsx

```diff
@@ -78,6 +79,25 @@ export const SqlResultTabs = observer<Props>(function SqlDataResult({ state, onT
     return true;
   }
 
+  function handleReorder(draggedTabId: string, targetTabId: string, position: 'before' | 'after') {
+    const draggedTab = state.tabs.find(t => t.id === draggedTabId);
+    const targetTab = state.tabs.find(t => t.id === targetTabId);
+
+    if (!draggedTab || !targetTab) {
+      return;
+    }
+
+    const reorderedTabs = reorderArray(state.tabs, draggedTab, {
+      item: targetTab,
+      position,
+    });
+
+    state.tabs = reorderedTabs.map((tab, index) => ({
+      ...tab,
+      order: index,
+    }));
```

Comment:
This is quite inefficient, `reorderArray` will mutate `state.tabs`, which will lead to re-render
`state.tabs = ...map(() => ({...}))` will mutate again and re-create every tab state, and this together will lead to a full re-render of any related data, while it's only  required to update `order` of changed tabs